vamp-server/convert.cpp

```diff
@@ -556,8 +558,15 @@ readInput(string format, RequestOrResponse::Direction direction, bool &eof)
     if (format == "json") {
         string err;
         auto result = readInputJson(direction, err, eof);
-        if (err != "") throw runtime_error(err);
-        else return result;
+        const bool isRecognisedError = !result.success && result.errorText != "";
+        
+        // if the RequestOrResponse (result) has been populated with success=false and an error message
+        // then the server returned a well formed error, it is safe to return it for conversion 
+        // -- but if err is populated, something else has gone wrong
+        if (isRecognisedError || err == "") 
```

Comment:
Agreed that reviewing at this point in the evening is a poor idea, better tomorrow (and this will not at all be a big deal then; it was never really supposed to be).

For what it's worth, simply changing the comment is fine to me, so long as it reflects having thought about the logic and decided that it was right already.